dts/bindings/wifi/nordic,nrf700x-tx-power.yaml

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: This is a representation of the nRF70 series transmit power ceilings
+  i.e., maximum transmit power allowed.
+
+compatible: "nordic,nrf700x-tx-power-ceiling"
+
+include: base.yaml
+
+properties:
+  max-pwr-2g-dsss:
+    type: int
+    required: true
+    description: 8-bit values for max tx power in 2.4GHz band for DSSS/CCK rates
+  max-pwr-2g-mcs0:
+    type: int
+    required: true
+    description: 8-bit values for max tx power in 2.4GHz band for MCS0
+  max-pwr-2g-mcs7:
+    type: int
+    required: true
+    description: 8-bit values for max tx power in 2.4GHz band for MCS7
+  max-pwr-5g-low-mcs0:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS0 low sub-band
+  max-pwr-5g-low-mcs7:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS7 low sub-band
+  max-pwr-5g-mid-mcs0:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS0 mid sub-band
+  max-pwr-5g-mid-mcs7:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS7 mid sub-band
+  max-pwr-5g-high-mcs0:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS0 high sub-band
+  max-pwr-5g-high-mcs7:
+    type: int
+    description: 8-bit values for max tx power in 5GHz band MCS7 high sub-band
```

Comment:
This is not modelling hardware, these are settings which should be Kconfigs IMO

Reply:
No, these are PCB/board dependent i.e., rely on calibration settings, so, part of the DTS, esp. for custom designs.

Reply:
Ok, I see linux has these limits in the dts as well :) See
https://github.com/torvalds/linux/blob/a19ea421490dcc45c9f78145bb2703ac5d373b28/Documentation/devicetree/bindings/net/wireless/mediatek%2Cmt76.yaml#L134

However, these properties should be part of the included by the base binding `nordic,nrf700x` rather than a standalone node, optimally we could get ahead and create a base binding for all wifi modems, like `wifi-power-limits.yaml` to be included by other wifi bindings :) Otherwise we may end up copying a lot of properties :)

Reply:
An additional question out of curiosity, why are some of the properties `required` and others not?

Reply:
> optimally we could get ahead and create a base binding for all wifi modems, like wifi-power-limits.yaml to be included by other wifi bindings :) 

Well, this was specific to Nordic, but if we have another user then can move to a base binding, don't want to do that without any other user (YAGNI).

> An additional question out of curiosity, why are some of the properties required and others not?

nRF7002 Wi-Fi has variant that supports 2.4GHz band only (nrf7001), the support for which will come in a later PR, so, all 5GHz was made optional to support both the variants.

Reply:
Ah, in that case the compatible is being misused here :) The compatible describes the exact hardware model, like `nordic,nrf7001` or `nordic,nrf7002`, device drivers can be generic and support both, but the compatible and accompanied bindings are tied to the specific hardware.

This ensures the DTS properties are valid for the specific hardware, if the wifi modem is `nordic,nrf7001`, `max-pwr-2g-dsss` is in the bindings file required, `max-pwr-5g-low-mcs0` is not, since the hardware does not support it :)

Reply:
> This ensures the DTS properties are valid for the specific hardware, if the wifi modem is nordic,nrf7001, max-pwr-2g-dsss is required, max-pwr-5g-low-mcs0 is not, since the hardware does not support it :) For nordic,nrf7002, both are present and required.

Yes, that is correct.

> Ah, in that case the compatible is being misused here :) The compatible describes the exact hardware model, like nordic,nrf7001 or nordic,nrf7002, device drivers can be generic and support both, but the compatible and accompanied bindings are tied to the specific hardware.

So, options are to split it to per-HW base or remove compatible and use it as a generic properties?

Reply:
> So, options are to split it to per-HW base or remove compatible and use it as a generic properties?

Sounds right, the generic properties would be included by the bindings for the specific hardware, like `spi-device.yaml`